### ISSUE TITLE
Subdomain wildcard CORS

### ISSUE THREAD

**Issue by davedray** [traefik/traefik#6851]
### Do you want to request a *feature* or report a *bug*?
Feature

Feature

### What did you expect to see?

I would like to be able to define a wildcard subdomain for CORS:

```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: testHeader
spec:
  headers:
    accessControlAllowMethods:
      - "GET"
      - "OPTIONS"
      - "PUT"
    accessControlAllowOriginList:
      - "https://*.bar.org"
    accessControlMaxAge: 100
    addVaryHeader: "true"
```

and then have it allow `foo.bar.org` as well as `baz.bar.org` for example.

We have a scenario where we are allowing vanity urls for certain clients, and they all hit the same API.  We either have to move CORS back into our api and out of traefik, do something hacky with forward auth, or (ideally) solve it in Traefik, but I don't want to work on the PR if it's unlikely to get merged in for whatever reason.

If it's a feature that (assuming the code was fine, of course) would be merged in I'm happy to do the work.

**Comment by dtomcej** [comment]
I think that this would be a welcome feature for people that have corporate domains that may serve hundreds of subdomains.

Manually keeping an up-to date list could be challenging.

Please feel free to open a PR.

**Comment by barthez** [comment]
I was considering similar report, but I would rather handle regular expressions than wildcards. There is a question how should the configuration look like, is it going to be a list of strings that headers middleware would treat as wildcards/regexps or there should be a way to provide a type of allowed origin matcher:

```
accessControlAllowMethods:
  - "http://plain.string.origin.com"
  - type: wildcard
    value: "http://*.wildcard.origin.com"
  - type: regexp
    value: "\Ahttps?://(subdomains|list)\.domain\.net\z"
```

**Comment by jodosha** [comment]
@davedray @dtomcej I followed @barthez suggestion and created a PR to allow regular expressions to be used here: https://github.com/containous/traefik/pull/6881 Please let me know what you think there.

**Comment by almereyda** [comment]
Is it possible that #6881 validated this ticket, which could be closed?

There is documentation about the feature, dubbed `accessControlAllowOriginListRegex`, in:

- https://doc.traefik.io/traefik/middlewares/http/headers/#accesscontrolalloworiginlistregex

**Comment by rtribotte** [comment]
Hello @almereyda,

Thanks, you are right, we missed closing it when we merged #6881.

Closed by #6881
